### PR TITLE
 allow configuration via a db url

### PR DESCRIPTION
**Pull request by sgargan**
I've been using Jennifer with some amber projects. The current generated database config in an amber project is in a url format (which is easily passed to crystal-db). I found myself parsing this into config for Jennifer in each and figured it would be nice to have Jennifer do this automatically.

### CODE REVIEW

File: spec/spec_helper.cr [hunk 1]
```diff
@@ -61,15 +61,22 @@ end
 
 Spec.before_each do
   Jennifer::Adapter.adapter.begin_transaction
+  reset_config
```
Comment (imdrasil) [inline spec/spec_helper.cr]: Why we do this twice (before and after)

File: src/jennifer/config.cr [hunk 2]
```diff
@@ -77,6 +89,48 @@ module Jennifer
...
+
+    def self.from_uri(uri : URI)
+      config.adapter = uri.scheme.to_s if uri.scheme
+      config.host = uri.host.to_s if uri.host
+      config.port = uri.port.not_nil!  if uri.port
+      config.db = uri.path.to_s.lchop if uri.path
+      config.user = uri.user.to_s if uri.user
+      config.password = uri.password.to_s if uri.password
+
+      if(uri.query)
```
Comment (imdrasil) [inline src/jennifer/config.cr]: Please remove `()`

File: src/jennifer/config.cr [hunk 3]
```diff
@@ -77,6 +89,48 @@ module Jennifer
...
+        params = HTTP::Params.parse(uri.query.to_s)
+        {% for field in CONNECTION_URI_PARAMS %}
```
Comment (imdrasil) [inline src/jennifer/config.cr]: Array literal as well has method `includes?` so I prefer to switch to it

Comment (imdrasil) [reply src/jennifer/config.cr]: Or we can use `&` method here - not sure which one will be better

File: spec/spec_helper.cr (hunk 1 above)
Comment (sgargan) [reply spec/spec_helper.cr]: The config tests manipulate the config state which could bleed into other tests. The second call in after was just to be ultra sure that the system is always consistent, in a known configuration state before the test and reset to a known config state after. I guess as there's only one set of hooks the before hook might be enough as its run before each test.

**Pull request merged**